Code/Framework/AzToolsFramework/AzToolsFramework/ActionManager/Menu/MenuManager.cpp

```diff
@@ -103,6 +121,28 @@ namespace AzToolsFramework
         return AZ::Success();
     }
 
+    MenuManagerOperationResult MenuManager::AddMenuToMenuBar(const AZStd::string& menuBarIdentifier, const AZStd::string& menuIdentifier, int sortIndex)
+    {
+        if (!m_menuBars.contains(menuBarIdentifier))
+        {
+            return AZ::Failure(AZStd::string::format(
+                "Menu Manager - Could not add menu \"%s\" to menu bar \"%s\" - menu bar has not been registered.", menuIdentifier.c_str(),
+                menuBarIdentifier.c_str()));
+        }
+
+        if (!m_menus.contains(menuIdentifier))
+        {
+            return AZ::Failure(AZStd::string::format(
+                "Menu Manager - Could not add menu \"%s\" to menu bar \"%s\" - menu has not been registered.", menuIdentifier.c_str(),
+                menuBarIdentifier.c_str()));
+        }
+
+        m_menuBars[menuBarIdentifier].AddMenu(sortIndex, menuIdentifier);
```

Comment:
update: It is going to be faster to perform a `find` first, and check if it is != to `end()`, than call `contains` and then use the `[]` operator to look-up the value. This is unfortunately doubling up the work, prefer...

```
auto menuBarIt = m_menuBars.find(menuBarIdentifier);
if (menuBarIt == m_menuBars.end())
{
    // error
}

menuBarIt->AddMenu(sortIndex, menuIdentifier);
```